### DISCUSSION TITLE
Query data frame without a copy

### DISCUSSION THREAD

**Discussion by djouallah** [duckdb/duckdb#4772] [General]
I shared this on twitter and people seems to like a lot 

```
import duckdb
con = duckdb.connect()
metadata = con.execute("SELECT * FROM parquet_metadata('lineitem_sorted.parquet')").df()
filter  = con.execute('''
                      SELECT path_in_schema,row_group_id,stats_min_value,stats_max_value  FROM metadata
                      where path_in_schema = 'l_shipdate'
                      and (stats_min_value <= '1994-01-01' and stats_max_value >= '1994-01-01')
                      ''').df()
filter
```

is there a way to use df using lazy evaluation, if we don't want to copy the data ?  maybe something like this ?

metadata = con.execute("SELECT * FROM parquet_metadata('lineitem_sorted.parquet')").to_arrow_dataset()

**Comment by hannes** [comment]
Which copy exactly did you want to avoid?

**Reply by djouallah** [reply]
I thought metadata df take some memory size  ?

**Comment by Alex-Monahan** [comment]
There is a PR to allow this with the relational API I believe! You will be able to chain together `.query()` calls!
#4471
